packages/expo-video/android/src/main/java/expo/modules/video/VideoPlayer.kt

```diff
@@ -24,12 +24,13 @@ import expo.modules.kotlin.sharedobjects.SharedObject
 import expo.modules.video.enums.PlayerStatus
 import expo.modules.video.enums.PlayerStatus.*
 import expo.modules.video.records.PlaybackError
+import expo.modules.video.records.VideoSource
 import expo.modules.video.records.VolumeEvent
 import kotlinx.coroutines.launch
 
 // https://developer.android.com/guide/topics/media/media3/getting-started/migration-guide#improvements_in_media3
 @UnstableApi
-class VideoPlayer(context: Context, appContext: AppContext, private val mediaItem: MediaItem) : AutoCloseable, SharedObject(appContext) {
+class VideoPlayer(context: Context, appContext: AppContext, private val mediaItem: MediaItem, source: VideoSource?) : AutoCloseable, SharedObject(appContext) {
```

Comment:
Can you please clarify the correlation between the `mediaItem` and the source? I'm unsure why we need both and what happens in the source was changed, but the media item wasn't.

Reply:
You were right to be confused, this was not necessary. It was enough to pass only the videoSource into the VideoPlayer constructor :)